### PR TITLE
java/python: added support for the "arrow" syntax

### LINKED ISSUE #3105: add support for the "arrow" syntax for java/python
When adding support for the librdkafka based kafka-c() destination, we introduced a new format for specifying name=>value entries that describe the librdkafka configuration. This is the arrow syntax, e.g:

```
                      config( "queue.buffering.max.ms" => "1000",
                             "message.timeout.ms" => "5000",
                             "debug" => "all")
```

This is now specific to kafka, but we have _very_ similar use-cases in java/python where we are passing options to the interpreted code, e.g.

scl/elasticsearch:
```
    option("index", `index`)
    option("type", `type`)
    option("server", `server`)
    option("port", "`port`")
    option("message-template", `template`)
    option("cluster", "`cluster`")
    option("flush_limit", `flush_limit`)
    option("flush_timeout", `flush_timeout`)
    option("client_mode", `client_mode`)
    option("resource", "`resource`")
    option("custom_id", "`custom_id`")
    option("concurrent_requests", `concurrent_requests`)
    option("skip_cluster_health_check", "`skip_cluster_health_check`")
    option("cluster_url", "`cluster_url`")
    option("java_keystore_filepath", "`java_keystore_filepath`")
    option("java_keystore_password", "`java_keystore_password`")
    option("java_truststore_filepath", "`java_truststore_filepath`")
    option("java_truststore_password", "`java_truststore_password`")
    option("java_ssl_insecure", `java_ssl_insecure`)
    option("http_auth_type", `http_auth_type`)
    option("http_auth_type_basic_username", "`http_auth_type_basic_username`")
    option("http_auth_type_basic_password", "`http_auth_type_basic_password`")
```

The goal would be to unify the syntax for these options in python/java, that is:

```
java/python(
     options("key" => "value",
                  "key2" => "value2"))
```

Couple of notes:
  * the keyword that introduces the config block should be plural (e.g. options instead of option)
  * there should be support for _multiple_ options blocks to make the life of wrapper SCLs easier (e.g. the java/python source/destination driver should be able to merge options specified via multiple options blocks).
  * we should be backward compatible in syntax (e.g. accept the old way of doing things)
  * we should be using common code for the implementation (see modules/kafka/kafka-grammar.ym as a starting point).

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
resolves #3105

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: modules/java/native/java-parser.c [hunk 1]
```diff
@@ -32,7 +32,7 @@ static CfgLexerKeyword java_keywords[] =
   { "java",        KW_JAVA },
   { "class_path",  KW_CLASS_PATH},
   { "class_name",  KW_CLASS_NAME},
-  { "option",      KW_OPTION},
+  { "options",     KW_OPTIONS},
```
Comment (alltilla) [inline modules/java/native/java-parser.c]: This breaks the user interface. Introducing `options` is great, but we must support the old syntax, too. See: `KWS_OBSOLETE`.

**Comment by furiel** [comment]
@mehul-m-prajapati did you have time to address @alltilla 's comment? I think only that is missing for this PR to be merged. Please let me know if you need clarification or if you got stuck.

File: modules/java/native/java-parser.c (hunk 1 above)
Comment (mehul-m-prajapati) [reply modules/java/native/java-parser.c]: @alltilla : okay. I have added KWS_OBSOLETE for option.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by mehul-m-prajapati** [comment]
> @mehul-m-prajapati did you have time to address @alltilla 's comment? I think only that is missing for this PR to be merged. Please let me know if you need clarification or if you got stuck.

@furiel : Yes, I have made the changes. You can review it now.

**Comment by alltilla** [comment]
@mehul-m-prajapati Perfect, thank you! :)

Can you write a quick newsfile entry in the news folder?
https://github.com/syslog-ng/syslog-ng/blob/master/news/README.md
This change categorizes as a `feature`.
Thanks. Cheers!

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**